tests/memleak/test.memleak.js

```diff
@@ -0,0 +1,458 @@
+'use strict';
+
+require('chai').should();
+
+var default_opts = {
+  /* This module can work in regular "test" mode, but it may also dump
+   * heap so that you can analyze what is going wrong.
+   * In the later case, simply set the following variable to `true`.
+   */
+  dump_snapshots: process.env.DUMP_HEAP === 'true' ? true : false,
+
+  /* How much growth is too much growth?
+   * In practice I can't get Node.js to not leak even with the simple
+   * classes defined below (maybe they are just plain buggy, but most
+   * of what heapdump reveals leads to nextTickQueue, so I suspect
+   * this is inherent to the way the tests are ran).
+   * Therefor we define the following as the maximum size and maximum
+   * percentage growth we will accept as "normal".
+   */
+  max_growth: 50000, // Up to 50ko lost in heap
+  max_percent: 1, // Up to 1% lost in heap
+
+  /* How many times should the test run? */
+  runs: 10000
+};
+
+var strict_opts = {
+  dump_snapshots: default_opts.dump_snapshots,
+  max_growth: 0,
+  max_percent: 0,
+  runs: 10000
+};
+
+/* A dummy adapter for test purposes. */
+
+function DummyPouchPlugin(PouchDB) {
+
+  function DummyPouchAdapter(opts,callback) {
+    var api = this;
+
+    api.close = function DummyPouchAdapterClose(callback) {
+      this.emit('closed');
+      if(callback) {
+        callback(null,null);
+      } else {
+        return Promise.resolve();
+      }
+    };
+    api.info = function DummyPouchAdapterInfo(callback) {
+      if(callback) {
+        callback(null,{dummy:true});
+      } else {
+        return Promise.resolve({dummy:true});
+      }
+    };
+
+    if(callback) {
```

Comment:
I need to put a fix in for eslint, but in general we prefer to have a space before parens

    if (callback) 
    api.info = function DummyPouchAdapterInfo (callback) {

etc, not a huge deal if you dont fix, will probably just do a batch fix when we fix eslint